File: web/src/app/func/Type.ts
/**
 * 类型实体
 */
export class Type {
  /** id */
  id: number;

  /** 名称 */
  name: string;

  deleted: boolean;

  constructor(data?: { id?: number, name?: string, deleted?: boolean}) {
    if (data) {
      if (data.id) {
        this.id = data.id;
      }

      if (data.name) {
        this.name = data.name;
      }

      if (data.deleted) {
        this.deleted = data.deleted;
      }
    }

  }
}

File: api/src/main/java/equipmentManagementSystem/entity/Type.java
package equipmentManagementSystem.entity;

import com.mengyunzhi.core.entity.YunzhiEntity;
import org.hibernate.annotations.SQLDelete;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.io.Serializable;

@Entity
@SQLDelete(sql = "update `type` set deleted = 1 where id = ?")
public class Type implements Serializable,SoftDelete {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;
    /**
     * 设备类型名称
     */
    private String name;

    private Boolean deleted = false;

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public Boolean getDeleted() {
        return this.deleted;
    }

    // 设置为私有
    private void setDeleted(Boolean deleted) {
        this.deleted = deleted;
    }
}
